### PR TITLE
Add impactless update command to RT

### PR DESCRIPTION
**Pull request by jhand2**
1. Add impactless update mailbox command
2. Fix a keyvault emulation bug which was clearing usage bits
3. Make ROM use the old FHT on update reset

### CODE REVIEW

**Review by jlmahowa-amd — APPROVED** [review]

File: rom/dev/src/fht.rs [hunk 1]
```diff
@@ -176,3 +178,13 @@ pub fn store(fht: FirmwareHandoffTable) {
     caliptra_common::print_fht(&fht);
     slice.copy_from_slice(fht.as_bytes());
 }
+
+pub fn load_from_dccm() -> CaliptraResult<FirmwareHandoffTable> {
+    let slice = unsafe {
```
Comment (rusty1968) [inline rom/dev/src/fht.rs:183]: cant we use the function try_load in common?

Comment (jhand2) [reply rom/dev/src/fht.rs]: Done.

File: drivers/src/mailbox.rs [hunk 2]
```diff
@@ -347,6 +347,19 @@ impl MailboxRecvTxn<'_> {
         Ok(())
     }
 
+    /// Pulls at most `count` words from the mailbox and throws them away
+    pub fn drop_words(&mut self, count: usize) -> CaliptraResult<()> {
```
Comment (rusty1968) [inline drivers/src/mailbox.rs:351]: create an issue to add a unit test for this later?

File: rom/dev/src/rom.ld [hunk 3]
```diff
@@ -19,7 +19,8 @@ ENTRY(_start)
 ROM_ORG          = 0x00000000;
 ICCM_ORG         = 0x40000000;
 DCCM_ORG         = 0x50000000;
-DATA_ORG         = 0x50004000;
+/* Avoid overlap with other long-lived DCCM structures */
+DATA_ORG         = 0x50004400;
```
Comment (rusty1968) [inline rom/dev/src/rom.ld:23]: not related to this PR, but create an issue to generate rom.ld from memory_layout.rs information.

File: rom/dev/src/rom.ld [hunk 4]
```diff
@@ -19,7 +19,8 @@ ENTRY(_start)
 ROM_ORG          = 0x00000000;
 ICCM_ORG         = 0x40000000;
 DCCM_ORG         = 0x50000000;
-DATA_ORG         = 0x50004000;
+/* Avoid overlap with other long-lived DCCM structures */
```
Comment (mhatrevi) [inline rom/dev/src/rom.ld:22]: Please update memory_layout.rs in common\src with this change.

File: rom/dev/src/flow/update_reset.rs [hunk 5]
```diff
@@ -81,9 +81,14 @@ impl UpdateResetFlow {
         Self::copy_regions();
         report_boot_status(UpdateResetOverwriteManifestComplete.into());
 
+        // Return the old FHT from DCCM
+        let fht = FirmwareHandoffTable::try_load()
```
Comment (mhatrevi) [inline rom/dev/src/flow/update_reset.rs:85]: Can the load and entry point addresses of RT be different than the previous in the new FW bundle? If yes, the values of these in FHT would be different from the actual load address of the new RT in ICCM.

**Review by mhatrevi — CHANGES_REQUESTED** [review]

File: rom/dev/src/flow/update_reset.rs (hunk 5 above)
Comment (jhand2) [reply rom/dev/src/flow/update_reset.rs:85]: Hmm. The ICCM address could change, but I don't think the DataVault handle can change. So we probably don't need to update the FHT. I'll update this.

However, we should be overwriting the load address in DataVault. But when I try to call `write_warm_reset_entry4(WarmResetEntry4::RtLoadAddr, manifest.runtime.load_addr)` during update reset, I get a Load/Store fault. I would expect these datavault entries to be unlocked. Maybe a simulator issue?

File: rom/dev/src/rom.ld (hunk 4 above)
Comment (jhand2) [reply rom/dev/src/rom.ld:22]: Actually I don't think this change is needed anymore. I don't see any overlap. I'll remove this.

File: rom/dev/src/rom.ld (hunk 3 above)
Comment (jhand2) [reply rom/dev/src/rom.ld:23]: This change is no longer needed. Removing.

File: drivers/src/mailbox.rs (hunk 2 above)
Comment (jhand2) [reply drivers/src/mailbox.rs:351]: Added a unit test

File: rom/dev/src/flow/update_reset.rs (hunk 5 above)
Comment (jhand2) [reply rom/dev/src/flow/update_reset.rs]: Changed to just not change FHT.

For the second part of the comment, opened https://github.com/chipsalliance/caliptra-sw/issues/518

File: rom/dev/src/rom.ld [hunk 6]
```diff
@@ -33,7 +33,7 @@ ROM_RELAXATION_PADDING = 4k;
 ROM_SIZE          = 32K;
 ICCM_SIZE         = 128K;
 DCCM_SIZE         = 128K;
-DATA_SIZE         = 96K;
+DATA_SIZE         = 95K;
```
Comment (mhatrevi) [inline rom/dev/src/rom.ld:36]: This change is no longer needed, right?

File: rom/dev/src/flow/warm_reset.rs [hunk 7]
```diff
@@ -25,11 +25,11 @@ impl WarmResetFlow {
     ///
     /// * `env` - ROM Environment
     #[inline(never)]
-    pub fn run(env: &mut RomEnv) -> CaliptraResult<FirmwareHandoffTable> {
+    pub fn run(_env: &mut RomEnv) -> CaliptraResult<Option<FirmwareHandoffTable>> {
         cprintln!("[warm-reset] ++");
 
         cprintln!("[warm-reset] --");
```
Comment (mhatrevi) [inline rom/dev/src/flow/warm_reset.rs:32]: There will be a conflict here due to https://github.com/chipsalliance/caliptra-sw/pull/516. Suggest keeping the workaround and not returning the FHT (this change).

**Review by mhatrevi — CHANGES_REQUESTED** [review]

File: rom/dev/src/rom.ld (hunk 6 above)
Comment (jhand2) [reply rom/dev/src/rom.ld]: Done.

File: rom/dev/src/flow/warm_reset.rs (hunk 7 above)
Comment (jhand2) [reply rom/dev/src/flow/warm_reset.rs:51]: Done.

**Review by mhatrevi — APPROVED** [review]

**Pull request merged**